azure-configurationservice/tests/test_azure_configuration_client.py

```diff
@@ -0,0 +1,601 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import datetime
+from copy import copy
+import pytest
+
+
+from msrestazure.azure_exceptions import CloudError
+from azure.configuration import AzureConfigurationClient
+from azure.configuration import ConfigurationSetting
+from devtools_testutils import AzureMgmtTestCase
+
+import conftest
+
+
+class AzConfigTestData:
+    def __init__(self):
+        self.key_uuid = None
+        self.label1 = None
+        self.label2 = None  # contains reserved chars *,\
+        self.label_uuid = None
+        self.label1_data = []
+        self.label2_data = []
+        self.no_label_data = []
+
+
+class AzConfigurationClientTest(AzureMgmtTestCase):
+    def setUp(self):
+        self.working_folder = os.path.dirname(__file__)
+        super(AzConfigurationClientTest, self).setUp()
+
+        connection_str = os.environ["AZ_CONFIG_CONNECTION"]
+        self.app_config_client = AzureConfigurationClient(connection_str)
+        self.test_data = conftest.setup_data()
+
+    def tearDown(self):
+        conftest.teardown_data(self.test_data)
+
+    # method: add_configuration_setting
+    def test_add_configuration_setting(self):
+
+        kv = ConfigurationSetting()
+        kv.key = "unit_test_key_add_" + self.test_data.key_uuid
```

Comment:
It is a little confusing why key and label come from somewhere else, and the rest are set locally

Reply:
self.test_data.key_uuid is the pre-generated uuid. Here I prefix it with "unit_test_key_add_". I do this because I want to use a uuid and the same uuid for every run. Again this is for replay.